### PR TITLE
Update docs around entrypoint option

### LINKED ISSUE #3140: entrypoint defined in docker-compose.yml wipes out CMD defined in Dockerfile
Hi, I am following the example on https://docs.docker.com/compose/startup-order/ to make sure the database is running before I start the application.
My `Dockerfile` contains the command

```
CMD ["/usr/bin/java", "-jar", "/usr/lib/gumtree/api-server/server/api-server.war"]
```

and in my `docker-compose.yml` I have 

```
entrypoint: ["/usr/bin/wait-for-it.sh", "postgres001:5432", "-t", "120", "--"]
```

but after the postgres database starts the service container just exits straight away.

```
api_1        | wait-for-it.sh: postgres001:5432 is available after 42 seconds
postgres001_1 | LOG:  database system is ready to accept connections
api_api_1 exited with code 0
```

Looking inside the running container I can see the entrypoint does not have the command appended

```
docker exec b456a362e587 ps aux
USER       PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root         1  0.2  0.0  21816  3200 ?        Ss   22:12   0:00 bash /usr/bin/wait-for-it.sh postgres001:5432 -t 120 --
```

If both command and entrypoint are in the same place (either in the `Dockerfile` or in the `docker-compose.yml`) the application starts up properly. 

```
› docker exec 6cafdb9fdcf2 ps aux
USER       PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root         1  0.2  0.0  21816  3208 ?        Ss   22:10   0:00 bash /usr/bin/wait-for-it.sh postgres001:5432 -t 120 -- java -jar /usr/lib/gumtree/api-server/server/api-server.war
```

Any idea is this is a bug or I am doing something wrong?

Thanks

### PR DESCRIPTION
**Pull request by aanand**
Closes #3140.

### CODE REVIEW

**Comment by joostmeijles** [comment]
Hi aanand,

I think there's still a small mistake in the wait-for-it.sh command. It should be:
`command: ["./wait-for-it.sh", "db:5432", "--", "python", "app.py"]`

**Comment by aanand** [comment]
@joostmeijles Thanks - fixed.

File: docs/compose-file.md [hunk 1]
```diff
@@ -272,6 +272,10 @@ The entrypoint can also be a list, in a manner similar to [dockerfile](https://d
         - memory_limit=-1
         - vendor/bin/phpunit
 
+> **Note**: Setting `entrypoint` will clear out any default command set on the
+> service's image (i.e. any `CMD` instruction in the Dockerfile used to build
```
Comment (SvenDowideit) [inline docs/compose-file.md]: only `CMD`? what about `ENTRYPOINT` ?

and do you mean "clear" or do you mean will over-ride?

File: docs/startup-order.md [hunk 2]
```diff
@@ -75,7 +74,7 @@ script:
         exec $cmd
 
     You can use this as a wrapper script as in the previous example, by setting
-    `entrypoint: ./wait-for-postgres.sh db`.
+    `command: ["./wait-for-postgres.sh", "db", "python", "app.py"]`.
```
Comment (SvenDowideit) [inline docs/startup-order.md]: so - what's the difference between using `entrypoint` and `command` in these examples - its a bit weird inthe context of the change in `compose-file.md`

**Comment by aanand** [comment]
Closed in favour of https://github.com/docker/docker.github.io/pull/712

**Pull request closed without merging**